client/go/outline/callback/callback.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2025 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package callback provides a thread-safe mechanism for managing and invoking callbacks.
+package callback
+
+import (
+	"log/slog"
+	"sync"
+)
+
+// Token can be used to uniquely identify a registered callback.
+type Token int
+
+// Callback is an interface that can be implemented to receive callbacks.
+//
+// It accepts an input and returns an output, allowing communication back to the caller.
+type Callback interface {
+	OnCall(data string) string
+}
+
+var (
+	mu        sync.RWMutex
+	callbacks       = make(map[Token]Callback)
```

Comment:
Same comment here. Create a callback map object, rather than using globals.

Reply:
Do you mean introduce a `CallbackManager` and then have a default singleton method such as `callback.DefaultManager()` ?